q2_taxa/plugin_setup.py

```diff
@@ -60,6 +60,116 @@
+                     'have a corresponding taxonomic annotation. Taxonomic '
+                     'annotations for features that are not present in the '
+                     'feature table will be ignored.'),
+        'table': 'Feature table to be filtered.'},
+    parameter_descriptions={
+        'include': ('One or more search terms that indicate which taxa should '
+                    'be included in the resulting table. If providing '
+                    'more than one term, terms should be delimited by the '
+                    'query-delimiter character. By default, all taxa '
+                    'will be included.'),
+        'exclude': ('One or more search terms that indicate which taxa should '
+                    'be excluded from the resulting table. If providing '
+                    'more than one term, terms should be delimited by the '
+                    'query-delimiter character. By default, no taxa '
+                    'will be excluded.'),
+        'mode': ('Mode for determining if a search term matches a taxonomic '
+                 'annotation. "contains" requires that the annotation '
+                 'has the term as a substring; "exact" requires that the '
+                 'annotation is a perfect match to a search term.'),
+        'query_delimiter': ('The string used to delimit multiple search terms '
+                            'provided to include or exclude. This parameter '
+                            'should only need to be modified if the default '
+                            'delimiter (a comma) is used in the provided '
+                            'taxonomic annotations.')
+    },
+    output_descriptions={
+        'filtered_table': ('The taxonomy-filtered feature table.')
+    },
+    name='Taxonomy-based feature table filter.',
+    description=('This method filters features from a table based on their '
+                 'taxonomic annotations. Features can be retained in the '
+                 'resulting table by specifying one or more include search '
+                 'terms, and can be filtered out of the resulting table by '
+                 'specifying one or more exclude search terms. If both '
+                 'include and exclude are provided, the inclusion critera '
+                 'will be applied before the exclusion critera. Either '
+                 'include or exclude terms (or both) must be provided.')
+)
+
+plugin.methods.register_function(
+    function=filter_seqs,
+    inputs={
+        'taxonomy': FeatureData[Taxonomy],
+        'sequences': FeatureData[Sequence]
+    },
+    parameters={'include': qiime2.plugin.Str,
+                'exclude': qiime2.plugin.Str,
+                'mode':
+                    qiime2.plugin.Str % qiime2.plugin.Choices(['exact',
+                                                               'contains']),
+                'query_delimiter': qiime2.plugin.Str},
+    outputs=[('filtered_sequences', FeatureData[Sequence])],
```

Comment:
Should this also return the sequences which were removed? Then it would act like [q2-quality-control does](https://github.com/qiime2/q2-quality-control/blob/c294ea77eb30cf80ed0bddc0cfde44ce730f8b22/q2_quality_control/plugin_setup.py#L37-L38). Then you you can inspect that the method did what you expected, and you should be able to use those new `FeatureData[Sequence]` artifacts to just filter your table (in either direction).

Reply:
Good suggestion, I'll add that in.

Reply:
@ebolyen and I discussed offline and decided not to add this in.